### ISSUE TITLE
GPL3.0

### ISSUE THREAD

**Issue by the-moog** [WangXuan95/FPGA-USB-Device#19]
Is this source code deiberately GPL3.0?

Only that means that, as it stands, this project can only be used in GPL3.0 compatible 'copy left' hardware as the act of synthisising it and loading into an FPGA is akin to a static link.
Giving both project owners and contributors equal access and rights to both sets of IP.  That makes using this in any commercial product almost impossible, even if not for profit, e.g. internal test equipment.
Would you consider adopting a similar but less restrictive license, e.g. Apache or MIT?  Or perhaps dual licence?

**Comment by lnchan** [comment]
If you'd look at issue #13, copyleft seems to be a deliberate choice, yes.

**Comment by the-moog** [comment]
@lnchan  I took a look at #13 and added (a rather verbose, for which I apologise.  I do ramble sometimes....) comment as an explanation to yourself and others why GPL3 can't work in  most FPGA projects.

**Comment by mangelajo** [comment]
Yes, GPL3 is a very hard license, it means that you need to share not only your library when used, but also any code that you compile together with your library.

I'd recommend something lighter light the MIT license, (i.e. https://github.com/SpinalHDL/VexRiscv/blob/master/LICENSE ) , Apache license, or similar.

But anyway, I would like to use the lines here to congratulate you for the impressive work you have done here @WangXuan95

Thank you for sharing with us all!.

**Comment by mangelajo** [comment]
Also look at LGPL,  this would help wider adoption of your library, while still ensuring that people will contribute back fixes and code to FPGA-USB-Device.

**Comment by perlindgren** [comment]
Please consider Apache2/MIT, it might be a showstopper for us to use together with our research oriented open source RISC-V. 
(https://github.com/perlindgren/hippomenes)

**Comment by jokinen086** [comment]
I also hope the license will be changed to Apache2/MIT.  I believe people want to use this great project easily, not with GPL3.0.
Thank you!

**Comment by WangXuan95** [comment]
@perlindgren  @jokinen086  @the-moog @mangelajo 

I've change the license to LGPL

**Comment by mangelajo** [comment]
Thank you @WangXuan95 your code will reach to many projects. And I hope that you will get good contributions back 👏👏

**Comment by jokinen086** [comment]
Thank you @WangXuan95 for changing the license. 
However, my understanding is that GPL and LGPL are essentially same in the FPGA design because dynamic link is NOT available.  What is the significant difference when changing to LGPL? Could you explain?
(I think I am familiar with open source license but I am not confident on how LGPL license can be applied to FPGA design.)
